include/cpr/accept_encoding.h

```diff
@@ -0,0 +1,40 @@
+#ifndef CPR_ACCEPT_ENCODING_H
+#define CPR_ACCEPT_ENCODING_H
+
+#include <curl/curlver.h>
+#include <initializer_list>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace cpr {
+
+enum class AcceptEncodingMethods {
+  identity,
+  deflate, 
+  zlib, 
+  gzip,
+};
+
+class AcceptEncoding {
+  public:
+    std::map<AcceptEncodingMethods, std::string> MethodsString { 
```

Comment:
Although I like the way you are solving this here, curl allows even more encodings: https://curl.se/libcurl/c/CURLOPT_ACCEPT_ENCODING.html

Since this is something that changes based on what is build into curl, in my eyes we need at least an additional option allowing the user to specify their own encoding via a string name.